core/vm/contracts_suave.go

```diff
@@ -133,3 +136,42 @@ type suaveRuntime struct {
 }
 
 var _ SuaveRuntime = &suaveRuntime{}
+
+func (s *suaveRuntime) doHTTPRequest(request types.HttpRequest) ([]byte, error) {
+	if request.Method != "GET" && request.Method != "POST" {
+		return nil, fmt.Errorf("only GET and POST methods are supported")
+	}
+	if request.Url == "" {
+		return nil, fmt.Errorf("url is empty")
+	}
+
+	var body io.Reader
+	if request.Body != nil {
+		body = bytes.NewReader(request.Body)
+	}
+
+	req, err := http.NewRequest(request.Method, request.Url, body)
```

Comment:
We probably want to enforce a timeout right?

Reply:
Who decides the timeout? Another field in the request?

Reply:
should we set a sane default? 5s? 2s? 10s?

Reply:
Added 5-second timeout as default for now.